command/fmt_test.go

```diff
@@ -50,3 +50,15 @@ func TestFmt_unfomattedTemlateDirectory(t *testing.T) {
 		fatalCommand(t, c.Meta)
 	}
 }
+
+func TestFmt_Recursive(t *testing.T) {
+	c := &FormatCommand{
+		Meta: testMeta(t),
+	}
+
+	args := []string{"-check=true", "-recursive=true", filepath.Join(testFixture("fmt"), "")}
+
+	if code := c.Run(args); code != 3 {
+		fatalCommand(t, c.Meta)
+	}
```

Comment:
Can you please check that the files were formatted recursively in this test as well ?

Reply:
@azr looking at the other existing test for fmt.go and formatter.go, they are not validating the contents of the files that are formatted. Looking at `filecheck`, this will validate the file but not undo the changes (cleanup on `filecheck` just deletes file, does not undo the changes). With this said, should I create another PR first to update the existing tests? It looks like adding this test functionality on top of recursive flag would be a lot for one PR. Please let me know if I'm missing existing functionality here or if I should go ahead with making the changes on another PR for increasing testing functionality before updating this PR with that.

Reply:
Wait I think I found solution, let me try to implement now.

Reply:
@azr Added the tests, please let me know if there's anything else I should add.

Reply:
love this ! Many thanks 🙂